tests/suite/test_transport_server_tcp_load_balance.py

```diff
@@ -407,3 +408,121 @@ def test_tcp_request_load_balanced_method(
             client.close()
 
         assert len(endpoints) is 3
+
+    @pytest.mark.skip_for_nginx_oss
+    def test_tcp_passing_healthcheck_with_match(
+            self, kube_apis, crd_ingress_controller, transport_server_setup, ingress_controller_prerequisites
+    ):
+        """
+        Configure a passing health check and check that all backend pods return responses.
+        """
+
+        # Step 1 - configure a passing health check
+
+        patch_src = f"{TEST_DATA}/transport-server-tcp-load-balance/passing-hc-transport-server.yaml"
+        patch_ts(
+            kube_apis.custom_objects,
+            transport_server_setup.name,
+            patch_src,
+            transport_server_setup.namespace,
+        )
+        # 4s includes 3s timeout for a health check to fail in case of a connection timeout to a backend pod
+        wait_before_test(4)
+
+        result_conf = get_ts_nginx_template_conf(
+            kube_apis.v1,
+            transport_server_setup.namespace,
+            transport_server_setup.name,
+            transport_server_setup.ingress_pod_name,
+            ingress_controller_prerequisites.namespace
+        )
+
+        match = f"match_ts_{transport_server_setup.namespace}_transport-server_tcp-app"
+
+        assert "health_check interval=5s port=3333" in result_conf
+        assert f"passes=1 jitter=0s fails=1 match={match}" in result_conf
+        assert "health_check_timeout 3s;"
+        assert 'send "health"' in result_conf
+        assert 'expect  "healthy"' in result_conf
+
+        # Step 2 - confirm load balancing works
```

Comment:
I would recommend not checking the load balancing here. I think the test should confirm the healthcheck functionality not the load balancing. If a test fails, it will be more accurate and the test easier to diagnose.

Reply:
In this test we check if  health checks that suppose to pass actually pass. For NGINX Plus, this means it will load balance to all 3 upstream servers, which NGINX Plus will consider healthy. If the health checks don't pass for any of upstream servers, NGINX Plus will not load balance to it. That's how we know there is a problem. Hope this clarifies.

Reply:
@soneillf5 @pleshakov looks like another "setting up clear boundaries" to mw, however I'm happy enough with just checking the contents in conf and making sure that reload passes.

Reply:
@pleshakov Ah I see now, the functionality is the change in load balancing behaviour. In which, yeah this is a great test and is fine as is.

Reply:
@vepatel @soneillf5 
I'm  happy to remove the load balancing part and leave out only the config check.

There is another way to check that health check pass or fail -- get the status information about the upstream servers from the API. Take a look at https://demo.nginx.com/#tcp_upstreams -- the health monitors group. That information is available via API.